### ISSUE TITLE
Display game sizes on the library page

### ISSUE THREAD

**Issue by Lrapava** [Heroic-Games-Launcher/HeroicGamesLauncher#937]
**Current behavior**
Currently, when you open the Library page and hover on a thumbnail a semi-transparent box appears with the title of the game. A bunch of space in the box is going to waste.

**Suggestion**
My suggestion is to display install size in the semi-transparent transparent box under the name of the game. That would help people who have limited storage to choose the game they can/want to install, since hovering on a thumbnail is easier than clicking, going back, clicking, going back and etc. The sizes can be calculated on startup.

**Comment by derrod** [comment]
There's no real way to know install size without fetching the manifest and calculating it. And that is quite slow.

**Comment by Lrapava** [comment]
> There's no real way to know install size without fetching the manifest and calculating it. And that is quite slow.

yeah, but they can be cached, right?

**Comment by flavioislima** [comment]
It is cached already but only after clicking on the game and opening the game page or the install modal.
Making it on startup or refresh will make heroic a lot slower. Like, really slow.

**Comment by Nocccer** [comment]
What about if we only show it if it was once fetched by clicking on gamepage or by clicked installmodal ?

**Comment by Lrapava** [comment]
> It is cached already but only after clicking on the game and opening the game page or the install modal. Making it on startup or refresh will make heroic a lot slower. Like, really slow.

I assume by "slow" you mean it'll take pages longer to load. Can't the sizes be calculated in the background/on a separate thread?

**Comment by yoshimo** [comment]
Size only updates when there is a new version or a new dlc released. So i think caching would be perfectly fine.
It should be fine if you update when a game page is loaded and crawl for the rest with multiple low priority threads to populate the cache if the entry is not available or out of date.

**Comment by flavioislima** [comment]
> > It is cached already but only after clicking on the game and opening the game page or the install modal. Making it on startup or refresh will make heroic a lot slower. Like, really slow.
> 
> I assume by "slow" you mean it'll take pages longer to load. Can't the sizes be calculated in the background/on a separate thread?

Even if we put this in background. When starting heroic for the first time it will need to download all info for all games. This uses Cpu and Ram. If you have a slow machine it can make heroic crash or even your system.

**Comment by flavioislima** [comment]
Wont do it
